egs/wsj/s5/steps/tfrnnlm/lstm.py

```diff
@@ -0,0 +1,328 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+#           2017 Hainan Xu, adapted to be used in Kaldi for lattice rescoring
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+# this script trains a vanilla RNNLM with TensorFlow. 
```

Comment:
I assume this script takes all the GPUs that it can see, if compiled for GPU?
On our queue, this is a problem; you have to use Gaurav's script  to get a suitable number of devices, and set CUDA_VISIBLE_DEVICES appropriately.  I've heard that TensorFlow does not actually make use of additional GPUs (beyond one) very efficiently, so please experiment and see whether we actually want to use more than one GPU.

Also you should explain in this script that it will use all visible GPUs and you have to set CUDA_VISIBLE_DEVICES (e.g. to "1", or "0,2", i.e. the subset of devices you want it to restrict its use to), if you don't want it to use all visible GPUs.  This is obvious to people who know TensorFlow but may not be obvious to Kaldi users.